RecoTracker/MkFit/plugins/MkFitGeometryESProducer.cc

```diff
@@ -66,6 +68,12 @@ class MkFitGeometryESProducer : public edm::ESProducer {
   void addTIDGeometry(mkfit::TrackerInfo &trk_info);
   void addTECGeometry(mkfit::TrackerInfo &trk_info);
 
+  void findRZBox(const GlobalPoint &gp, float &rmin, float &rmax, float &zmin, float &zmax);
+  void aggregateMaterialInfo(mkfit::TrackerInfo &trk_info);
+  void fillLayers(mkfit::TrackerInfo &trk_info);
+  std::vector<std::tuple<float, float, float>> material_histogram[mkfit::Config::nBinsZMat][mkfit::Config::nBinsRMat];
+  int neighbor_map[mkfit::Config::nBinsZMat][mkfit::Config::nBinsRMat];
```

Comment:
How big are these data structures?

They are used only temporarily in the creation of the `MkFitGeometry`, right? If they are large, it would be more economical to destruct them once the `MkFitGeometry` has been created instead of keeping them alive until the end of the job.

Reply:
Good point.
the grid size is 300x130 = 39K.
So, it will add up to around a MB. Still, better to cleanup.

Reply:
yes, i think i already commented on this before the final version :)

Reply:
this looks a bit ugly after passing the histogram through the add___Geometry functions  etc.